Jenkinsfile

```diff
@@ -137,7 +137,49 @@ pipeline {
         }
       }
     }
-  }
+    stage('Building packages') {
+      agent { label 'docker && linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/.local/bin"
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        dir("${BASE_DIR}"){
+          sh script: 'pip3 install --user cibuildwheel', label: "Installing cibuildwheel"
+          sh script: 'mkdir wheelhouse', label: "creating wheelhouse"
+          sh script: 'cibuildwheel --platform linux --output-dir wheelhouse; ls -l wheelhouse'
+          stash allowEmpty: true, name: 'packages', includes: "path_to_packages/**", useDefaultExcludes: false
+        }
+      }
+    }
+      stage('Release') {
+        agent { label 'linux && immutable' }
+        options { skipDefaultCheckout() }
+        environment {
+          HOME = "${env.WORKSPACE}"
+          PATH = "${env.PATH}:${env.WORKSPACE}/.local/bin"
+        }
+        when {
+          beforeAgent true
+          anyOf {
+            tag "v\\d+\\.\\d+\\.\\d+*"
+            expression { return params.Run_As_Master_Branch }
+          }
+        }
+        steps {
+          input(message: 'Should we release a new version?', ok: 'Yes, we should.')
+          deleteDir()
+          unstash 'source'
+          unstash('packages')
+          dir("${BASE_DIR}"){
+            /** TODO code to release */
```

Comment:
```suggestion
  def jsonValue = getVaultSecret(secret: 'secret')
  wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [
    [var: 'TWINE_USER', password: jsonValue.data.user],
    [var: 'TWINE_PASSWORD', password: jsonValue.data.password],
    ]]) {
    withEnv([
      "TWINE_USER=${jsonValue.data.user}",
      "TWINE_PASSWORD=${jsonValue.data.password}"]) {
        sh(label: "Release packages", script: """
        set +x
        python -m pip install twine 
        python setup.py sdist 
        python -u \${TWINE_USER} -p \${TWINE_PASSWORD} -m twine upload --skip-existing --repository-url https://upload.pypi.org/legacy/ dist/*.tar.gz 
        python -u \${TWINE_USER} -p \${TWINE_PASSWORD} -m twine upload --skip-existing --repository-url https://upload.pypi.org/legacy/ wheelhouse/*.whl
        """)
    }
  }            
```

Reply:
A couple of things:

 * we can't install Python modules globally due to access rights. So we need to use the `--user` flag:

        python3 -m pip install --user twine

 * the `-u` /  `-p` flags need to be given to `twine upload`, not `python`

        python3 -m twine upload -u \${TWINE_USER} -p \${TWINE_PASSWORD} --skip-existing --repository-url https://upload.pypi.org/legacy/ dist/*.tar.gz

Reply:
I've made the change